hs_file_types/views.py

```diff
@@ -348,6 +348,33 @@ def move_aggregation(request, resource_id, hs_file_type, file_type_id, tgt_path=
                 response_data['message'] = err_msg
                 return JsonResponse(response_data, status=status.HTTP_400_BAD_REQUEST)
 
+    file_override = request.POST.get('file_override', False)
+    if not isinstance(file_override, bool):
+        file_override = True if str(file_override).lower() == 'true' else False
+
+    # check if files already exist in the target path
+    res_files = []
+    override_tgt_paths = []
+    override_tgt_res_files = []
+    file_type_obj = FILE_TYPE_MAP[hs_file_type]
+    aggregation = file_type_obj.objects.get(id=file_type_id)
+    res_files.extend(aggregation.files.all())
+    istorage = res.get_irods_storage()
+    for file in res_files:
+        tgt_full_path = os.path.join(res.file_path, tgt_path, os.path.basename(file.storage_path))
+        if istorage.exists(tgt_full_path):
+            override_tgt_paths.append(tgt_full_path)
+            override_tgt_res_files.append(file)
+
+    if override_tgt_paths:
+        if not file_override:
+            override_file_names = ', '.join([os.path.basename(tgt_path) for tgt_path in override_tgt_paths])
+            message = f'aggregation move would overwrite {override_file_names}'
+            return HttpResponse(message, status=status.HTTP_300_MULTIPLE_CHOICES)
+        # delete conflicting files so that move can succeed
+        for override_file in override_tgt_res_files:
+            delete_resource_file_only(res, override_file)
```

Comment:
If the file being deleted here is part of an aggregation, it won't delete the aggregation and can corrupt the resource.

Reply:
I tested out moving an aggregation to a folder with the same aggregation, and the aggregation got moved successfully if override option is selected. It appears deleting conflicting resource files followed by moving the aggregation works. Could you give an example that could result in corrupt resource here? Thanks @pkdash

Reply:
You can test using the following steps:
1. Create a new resource and create a folder.
2. Upload a nc file to that folder which should create netcdf aggregation.
3. Upload the same nc file to the root of the resource. That should create another netcdf aggregation.
4. Move the netcdf aggregation at the root of the resource to the folder you created in step 1. When prompted to overwrite, select "Proceed".
5. In the UI it is not easy to detect if the netcdf aggregation originally inside the folder got deleted as we don't show the number of aggregation by types on the UI. However, if you download the bag you can see reference 2 netcdf aggregation. If you remove the content type for the netcdf aggregation that exists in the folder, you will still see in the UI that the resource still contains netcdf aggregation.